taurus_pyqtgraph/taurusplotdataitem.py

```diff
@@ -62,13 +62,14 @@ def __init__(self, *args, **kwargs):
                                     self.setXModel, 'XModel')
 
     def setXModel(self, xModel):
+        if self.xModel is not None:
```

Comment:
I don understand this logic. 
Shouldn't the listener be removed when xModel is None?
(I know that this was not the case before, but that was not necessarily correct)

Reply:
> Shouldn't the listener be removed when xModel is None?

The model should be removed if you set a valid model before. It does not matter if you set a valid model or None

Imaging that  you setXModel twice (model a/b/c/d and e/f/g/h), the previous listener must be removed (a/b/c/d) cause if you don't do it, you could have a TaurusPlotDataItem listening  two models  and you only would like to react to the event of the latest model (e/f/g/h).
So, if self.xModel is not None means that you call setXModel at least once, and you have been added a  listener.

Reply:
>  the previous listener must be removed (a/b/c/d) cause if you don't do it, you could have a TaurusPlotDataItem listening two models and you only would like to react to the event of the latest model (e/f/g/h).

I agree... but that does not answer my question. My concern is with the case in which the new model is `None`. 
To reuse your example, consider that you first do `setXModel('a/b/c/d')` and later on you do  `setXModel(None)`. 

IMHO, it makes sense to  **also** stop listening to a/b/c/d in this case.

Reply:
> IMHO, it makes sense to also stop listening to a/b/c/d in this case.

As I said: The listener must be stopped in both case.  If you check the code `self.xModel` is initialized  whit `None`, So, If  you set as first model `None` there is not listener to be removed.
But if you set `a/b/c/d/` model `self.xModel` is initialized few line below in this method. Then, if the next model is `None` or `e/f/g/h`, the current listener must be removed.

Reply:
> Then, if the next model is None or e/f/g/h, the current listener must be removed.

... therefore, the proposed code is *not* ok